lib/msal-browser/src/app/PublicClientApplication.ts

```diff
@@ -130,54 +130,62 @@ export class PublicClientApplication implements IPublicClientApplication {
         return this.handleRedirectResponse();
     }
 
-    /**
-     * Checks if navigateToLoginRequestUrl is set, and:
-     * - if true, performs logic to cache and navigate
-     * - if false, handles hash string and parses response
-     */
-    private async handleRedirectResponse(): Promise<AuthenticationResult | null> {
+    private getRedirectResponseHash(): string {
         // Get current location hash from window or cache.
         const { location: { hash } } = window;
         const isResponseHash = UrlString.hashContainsKnownProperties(hash);
+        const cachedHash = this.browserStorage.getItem(this.browserStorage.generateCacheKey(TemporaryCacheKeys.URL_HASH), CacheSchemaType.TEMPORARY) as string;
+        this.browserStorage.removeItem(this.browserStorage.generateCacheKey(TemporaryCacheKeys.URL_HASH));
 
-        // Check for interaction type
-        if (isResponseHash) {
+        const responseHash = isResponseHash ? hash : cachedHash;
+        if (responseHash) {
             // Deserialize hash fragment response parameters.
-            const hashUrlString: UrlString = new UrlString(hash);
-            const serverParams: ServerAuthorizationCodeResponse = hashUrlString.getDeserializedHash<ServerAuthorizationCodeResponse>();
+            const serverParams: ServerAuthorizationCodeResponse = StringUtils.queryStringToObject<ServerAuthorizationCodeResponse>(responseHash);
             const requestStateObj: RequestStateObject = ProtocolUtils.parseRequestState(this.browserCrypto, serverParams.state);
             const platformStateObj: BrowserStateObject = BrowserProtocolUtils.parseBrowserRequestState(this.browserCrypto, requestStateObj.libraryState.platformState);
-            if (StringUtils.isEmpty(platformStateObj.interactionType) || platformStateObj.interactionType !== InteractionType.REDIRECT) {
+            if (platformStateObj.interactionType !== InteractionType.REDIRECT) {
```

Comment:
Not sure why the empty check was removed, should probably add it back.

Reply:
It's redundant isn't it? `"" !== "redirect"` and `null !== "redirect"`